### PR TITLE
Add Credo

### PR DESCRIPTION
**Pull request by angelikatyborska**
This PR adds credo and fixes all problems reported by it. It also adds the credo check to the CI.

I'm also sneaking in `.tool-versions`, the same as the test runner and representer have.

### CODE REVIEW

File: .credo.exs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        excluded: [
+          ~r"/_build/",
+          ~r"/deps/",
+          ~r"/node_modules/",
+          # exclude exercise analysis because of snippets in `feature do form do...`
+          # they can be unusual, e.g. `_ignore || _ignore`
+          "lib/elixir_analyzer/test_suite/",
+          # exclude exercise analysis test because of example solutions used in tests
+          # those mimic student solutions and might be "bad code" on purpose
+          "test/elixir_analyzer/test_suite/"
```
Comment (angelikatyborska) [inline .credo.exs:43]: This is the only change in this config compared to the default config.

File: lib/elixir_analyzer/exercise_test.ex [hunk 2]
```diff
@@ -23,6 +23,7 @@ defmodule ElixirAnalyzer.ExerciseTest do
   #
 
   defmacro __before_compile__(env) do
+    # credo:disable-for-previous-line Credo.Check.Refactor.CyclomaticComplexity
```
Comment (angelikatyborska) [inline lib/elixir_analyzer/exercise_test.ex:26]: This is the only place where I was afraid of splitting this code to reduce cyclomatic complexity, so I took the easy way out...

File: .tool-versions [hunk 3]
```diff
@@ -0,0 +1,2 @@
+elixir 1.11.3-otp-23
+erlang 23.2.6
```
Comment (neenjaw) [inline .tool-versions:2]: 👍🏻 Looks good!

**Review by neenjaw — APPROVED** [review]
Looks good, forgot how opinionated credo is about `cond` vs `if`, but I'm good with it

**Pull request merged**